prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -1662,7 +1670,7 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         if is_mut {
             stmts.extend(self.encode_transfer_permissions(
                 lhs_place.clone(),
-                rhs_place,
+                rhs_place.unwrap(),
```

Comment:
`is_mut` implies `rhs_place.is_some()`?

Reply:
Yes. The tuple returned by `encode_loan_places` only returns `None` for `restored` when the `source` is a string literal (as it can't be encoded); and in the case of the string literal, `is_mut` is `False`. https://github.com/viperproject/prusti-dev/blob/71b66481f11fbc6eb1038a46d7e7271f5edcb8c6/prusti-viper/src/encoder/procedure_encoder.rs#L1458